tools/rimage/src/manifest.c

```diff
@@ -240,20 +240,21 @@ static int man_get_module_manifest(struct image *image, struct manifest_module *
 	segment->flags.r.load = 1;
 	segment->flags.r.readonly = 1;
 	segment->flags.r.code = 1;
+	segment->flags.r.type = SOF_MAN_SEGMENT_TEXT;
 
 	/* data segment */
 	segment = &man_module->segment[SOF_MAN_SEGMENT_RODATA];
 	segment->flags.r.contents = 1;
 	segment->flags.r.alloc = 1;
 	segment->flags.r.load = 1;
-	segment->flags.r.readonly = 1;
+	segment->flags.r.readonly = 0; /* rodata segment contains also writtable data */
 	segment->flags.r.data = 1;
-	segment->flags.r.type = 1;
+	segment->flags.r.type = SOF_MAN_SEGMENT_RODATA;
```

Comment:
How immutable is this ABI? Can it be extended to have both .data and .rodata? At least we could rename `SOF_MAN_SEGMENT_RODATA` to something that isn't confusing

Reply:
Hard to say. The rimage uses fixed segment indexes for specific type, module loading in zephyr is based on information about the segment type/flags without taking the index into account. The library manager ignores types/flags and assumes that the specified type is at a specific fixed index. In #8544 I modified the library manager to use segment type/flags instead of const index, so theoretically we can separate `.data` from `.rodata`, but the manifest only supports 3 segments! I don't know how the rom behaves, but I suspect that it was easier to implement flag support rather than constant indexes.

Reply:
would be good to clarify and hopefully fix this